projects/lib/src/board/board.cpp

```diff
@@ -529,14 +534,23 @@ bool Board::setFenString(const QString& fen)
 		if (square >= boardSize)
 			return false;
 
-		pieceStr.append(c);
-		Piece piece = pieceFromSymbol(pieceStr);
-		if (!piece.isValid())
-			continue;
+		// read ahead for multi-character symbols
+		int l, lsym_minus1 = 0;
+		for (l = 0; l < maxsymlen && i + l < token->length(); l++)
```

Comment:
It took me a while to understand this loop. To me it would be more logical to first try the maximum length token (because we have to try it eventually anyway), and if it doesn't match a piece symbol, then try `maxsymlen - 1` etc. Then we only have to call `setSquare()` when we really have the right piece, and we can break out of the loop early.